src/flask_container_scaffold/app_scaffold.py

```diff
@@ -2,76 +2,96 @@
 
 from flask import Flask
 
-from flask_container_scaffold.util import load_yaml
+from flask_container_scaffold.app_configurator import AppConfigurator
 
 
 class AppScaffold(object):
 
     def __init__(self, app=None,
-                 name=__name__, config=None):
+                 name=__name__, config=None,
+                 settings_required=False,
+                 instance_path=None,
+                 instance_relative_config=True):
         """
         This class provides a way to dynamically configure a Flask application.
 
-        :param app: An existing Flask application, if passed, otherwise we will
-                    create a new one
-        :param name: The name of the application, defaults to __name__.
-        :param config: A dict of configuration details. This can include
+        :param app obj: An existing Flask application, if passed, otherwise we
+                    will create a new one
+        :param name str: The name of the application, defaults to __name__.
+        :param config dict: A dict of configuration details. This can include
                        standard Flask configuration keys, like 'TESTING', or
                        custom keys (currently limited to a set list) to make
                        them available to the application during runtime
+        :param settings_required bool: Whether your app requires certain
+                        settings be specified in a settings.cfg file
+        :param instance_path str: Passthrough parameter to flask. An
+                    alternative instance path for the application. By default
+                    the folder 'instance' next to the package or module is
+                    assumed to be the instance path.
+        :param instance_relative_config bool: Passthrough parameter to flask.
+                    If set to True relative filenames for loading the config
+                    are assumed to be relative to the instance path instead of
+                    the application root.
 
         """
+        # TODO: Consider taking **kwargs here, so we can automatically support
+        # all params the flask object takes, and just pass them through.  Keep
+        # the ones we already have, as they are needed for the current code to
+        # work.
         Flask.jinja_options = dict(Flask.jinja_options, trim_blocks=True,
                                    lstrip_blocks=True)
-        self.app = app or Flask(name, instance_relative_config=True)
+        self.app = (app or
+                    Flask(name,
+                          instance_relative_config=instance_relative_config,
+                          instance_path=instance_path))
         self.config = config
+        self.silent = not settings_required
+        self.relative = instance_relative_config
         self._init_app()
 
     def _init_app(self):
-        self._setup_instance_folder()
-        self._configure_app()
+        self._load_flask_settings()
```

Comment:
Do we have an advantage to calling these methods here or directly in the __init__ method?

Reply:
It may not be a real 'advantage', I just like to separate simple instance variable assignment from more involved setup when I can

Reply:
That's fair. My only concern was the extra jump to `_init_app`. Splitting out the others are fine with me and what I prefer to do as well.

Reply:
I agree it is not a huge win right now, and perhaps not that necessary.  If this initialization does not get more complex in the near future, I am open to moving this into the main __init__ in a future patch.